src/SentrySdk.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+use Sentry\Integration\IntegrationInterface;
+use Sentry\State\Hub;
+use Sentry\State\HubInterface;
+use Sentry\State\Scope;
+
+/**
+ * This class is the main entry point for all the most common SDK features.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class SentrySdk
+{
+    /**
+     * @var HubInterface|null The current hub
+     */
+    private static $currentHub;
+
+    /**
+     * Constructor.
+     */
+    private function __construct()
+    {
+    }
+
+    /**
+     * Initializes the SDK by creating a new hub instance each time this method
+     * gets called.
+     *
+     * @return HubInterface
+     */
+    public static function init(): HubInterface
+    {
+        self::$currentHub = new Hub();
+
+        return self::$currentHub;
+    }
+
+    /**
+     * Gets the current hub. If it's not initialized then creates a new instance
+     * and sets it as current hub.
+     *
+     * @return HubInterface
+     *
+     * @deprecated
+     *
+     * @internal
+     */
+    public static function getCurrentHub(): HubInterface
+    {
+        @trigger_error(sprintf('The %s() method is deprecated since version 2.2 and will be removed in 3.0.', __METHOD__), E_USER_DEPRECATED);
+
+        if (null === self::$currentHub) {
+            self::$currentHub = new Hub();
+        }
+
+        return self::$currentHub;
+    }
+
+    /**
+     * Sets the current hub.
+     *
+     * @param HubInterface $hub
+     *
+     * @return HubInterface
+     *
+     * @deprecated
+     *
+     * @internal
+     */
+    public static function setCurrentHub(HubInterface $hub): HubInterface
+    {
+        @trigger_error(sprintf('The %s() method is deprecated since version 2.2 and will be removed in 3.0.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
same thing here

Reply:
The reason is that this method is needed to make the (also) deprecated method `Hub::setCurrentHub` work